source/data-modeling/field-behaviors.txt

```diff
@@ -0,0 +1,292 @@
+useful to save storage space, because MongoDB stores all field information along
+with every document.
+
+You can set an alternate storage name by using the ``as:`` keyword. The
+following example creates a field called ``name`` that {+odm+} stores in the database as
+``n``:
+
+.. literalinclude:: /includes/data-modeling/field-behaviors.rb
+   :language: ruby
+   :start-after: # start-field-as
+   :end-before: # end-field-as
+
+{+odm+} stores the ``name`` field as ``"n"``, but you can still access the field as
+``name`` in your application.
+
+Field Aliases
+-------------
+
+You can create an alias for your field by using the ``alias_attribute`` option.
+Specifying an alias does not change how {+odm+} stores the field in the
+database, but it allows you to access the field by a different name in your
+application. 
+
+The following example specifies an alias for the ``name`` field:
+
+.. literalinclude:: /includes/data-modeling/field-behaviors.rb
+   :language: ruby
+   :start-after: # start-field-alias
+   :end-before: # end-field-alias
+
+To remove a field alias, you can use the ``unalias_attribute`` option. The
+following example removes the alias for the ``name`` field:
+
+.. literalinclude:: /includes/data-modeling/field-behaviors.rb
+   :language: ruby
+   :start-after: # start-field-unalias
+   :end-before: # end-field-unalias
+
+You can also use ``unalias_attribute`` to remove the predefined ``id`` alias from the
+``_id`` field. This can be used to store different values in the ``_id`` field
+and an ``id`` field.
+
+Field Redefinition
+------------------
+
+By default, {+odm+} allows you to redefine fields on a model. To raise an error
+when a field is redefined, set the ``duplicate_fields_exception`` configuration
+option in your ``mongoid.yml`` file to ``true``.
+
+If the ``duplicate_fields_exception`` option is set to ``true``, you can still
+redefine a specific field by setting the ``overwrite`` option to ``true`` when
+you define the field. The following example defines the ``name`` field, and then
+redefines the field by using the ``overwrite`` option:
+
+.. literalinclude:: /includes/data-modeling/field-behaviors.rb
+   :language: ruby
+   :start-after: # start-field-overwrite
+   :end-before: # end-field-overwrite
+
+Custom ID Field
+---------------
+
+By default, {+odm+} defines the ``_id`` field on documents to contain a
+``BSON::ObjectId`` value that {+odm+} generates automatically. You can customize
+the type or the specify the default value of the ``_id`` field by specifying it in your
+model.
+
+The following example creates a ``Band`` class with a custom ``_id`` field:
+
+.. literalinclude:: /includes/data-modeling/field-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-id
+   :end-before: # end-custom-id
+
+You can omit the default value for the ``_id`` field. If you don't specify a
+default value for the field, {+odm+} persists the value without an ``_id``
```

Comment:
Is this meant to be document? Was a little hard to parse.
```suggestion
You can omit the default value for the ``_id`` field. If you don't specify a
default value for the field, {+odm+} persists the document without an ``_id``
```

Reply:
Oops, yeah that should be document